src/asymmetric_crypto_ctx.cpp

```diff
@@ -463,16 +463,87 @@ namespace mococrw {
+              _sigFormat{format}
+            {}
+
+        void verifyDigest(const std::vector<uint8_t> &signature,
+                          const std::vector<uint8_t> &messageDigest) {
+            if (_sigFormat == ECSDASignatureFormat::IEEE1363) {
+                size_t keySizeBytes = (key.getKeySize() + 7) / 8;
+                auto asn1Signature = _IEEE1363EcSignatureToAsn1ECSignature(signature, keySizeBytes);
+                _verifyAsn1(asn1Signature, messageDigest);
+            } else if (_sigFormat == ECSDASignatureFormat::ASN1_SEQUENCE_OF_INTS) {
+                _verifyAsn1(signature, messageDigest);
+            } else {
+                throw MoCOCrWException("ECDSA Signature type not recognized.");
+            }
+        }
+
+        void verifyMessage(const std::vector<uint8_t> &signature,
+                           const std::vector<uint8_t> &message) {
+            verifyDigest(signature, createHash(hashFunction, message));
+        }
+
+    private:
+        void _verifyAsn1(const std::vector<uint8_t> &signature,
+                         const std::vector<uint8_t> &messageDigest) {
+
+            size_t expectedDigestSize = Hash::getDigestSize(hashFunction);
+            if (messageDigest.size() != expectedDigestSize) {
+                throw MoCOCrWException((boost::format{"Expected digest of size %1%"}
+                                                    % expectedDigestSize).str());
+            }
+
+            try {
+                auto keyCtx = _EVP_PKEY_CTX_new(key.internal());
+                _EVP_PKEY_verify_init(keyCtx.get());
+
+                _EVP_PKEY_verify(keyCtx.get(),
+                                 reinterpret_cast<const unsigned char *>(signature.data()),
+                                 signature.size(),
+                                 reinterpret_cast<const unsigned char *>(messageDigest.data()),
+                                 messageDigest.size());
+            }
+            catch (const OpenSSLException &e) {
+                throw MoCOCrWException(e.what());
+            }
+        }
+
+        ECSDASignatureFormat _sigFormat;
     };
 
+    ECDSASignaturePublicKeyCtx::ECDSASignaturePublicKeyCtx(const AsymmetricPublicKey& key,
+                                                           openssl::DigestTypes hashFunction,
+                                                           ECSDASignatureFormat format)
+        : _impl(std::make_unique<ECDSASignaturePublicKeyCtx::Impl>(key, hashFunction, format)) {}
+
     ECDSASignaturePublicKeyCtx::ECDSASignaturePublicKeyCtx(const AsymmetricPublicKey& key,
                                                            openssl::DigestTypes hashFunction)
-        : _impl(std::make_unique<ECDSASignaturePublicKeyCtx::Impl>(key, hashFunction)) {}
+        : ECDSASignaturePublicKeyCtx(key, hashFunction, ECSDASignatureFormat::ASN1_SEQUENCE_OF_INTS) {}
 
     ECDSASignaturePublicKeyCtx::ECDSASignaturePublicKeyCtx(const X509Certificate& cert,
```

Comment:
One of the constructor definitions is missing here:

```
    ECDSASignaturePublicKeyCtx(const X509Certificate& cert, openssl::DigestTypes hashFunction,
                               ECSDASignatureFormat sigFormat);
```

Reply:
good catch.